lib/signaling/v2/room.js

```diff
@@ -166,10 +180,25 @@ RoomV2.prototype._update = function _update(roomState) {
     this.recording.update(roomState.recording);
   }
 
-  if (roomState.published) {
+  if (roomState.published && roomState.published.revision > this._publishedRevision) {
+    this._publishedRevision = roomState.published.revision;
+    roomState.published.tracks.forEach(function(track) {
+      if (track.sid) {
+        this._published.set(track.id, track.sid);
+      }
+    }, this);
     this.localParticipant.update(roomState.published);
   }
 
+  if (roomState.subscribed && roomState.subscribed.revision > this._subscribedRevision) {
+    this._subscribedRevision = roomState.subscribed.revision;
+    roomState.subscribed.tracks.forEach(function(track) {
```

Comment:
@markandrus Shouldn't we also remove any tracks that have been unsubscribed?

Reply:
Same rationale as above